storm-core/src/clj/org/apache/storm/daemon/nimbus.clj

```diff
@@ -1355,48 +1366,843 @@
+                                "totalMem"   (get (.get_total_resources sup) Config/SUPERVISOR_MEMORY_CAPACITY_MB)
+                                "totalCpu"   (get (.get_total_resources sup) Config/SUPERVISOR_CPU_CAPACITY)
+                                "usedMem"    (.get_used_mem sup)
+                                "usedCpu"    (.get_used_cpu sup)})
+            })
+         sups)))
+
+(defn send-cluster-metrics-to-executors [nimbus-service nimbus]
+  (let [cluster-summary (.getClusterInfo nimbus-service)
+        cluster-metrics (extract-cluster-metrics cluster-summary)
+        supervisors-metrics (extract-supervisors-metrics cluster-summary)]
+    (dofor
+      [consumer-executor (:cluster-consumer-executors nimbus)]
+      (do
+        (.handleDataPoints consumer-executor (:cluster-info cluster-metrics) (:data-points cluster-metrics))
+        (dofor
+          [supervisor-metrics supervisors-metrics]
+          (do
+          (log-message (:supervisor-info supervisor-metrics) " / " (:data-points supervisor-metrics))
+          (.handleDataPoints consumer-executor (:supervisor-info supervisor-metrics) (:data-points supervisor-metrics))))))))
+
+(defn mk-reified-nimbus [nimbus conf blob-store]
+  (let [principal-to-local (AuthUtils/GetPrincipalToLocalPlugin conf)
         admin-users (or (.get conf NIMBUS-ADMINS) [])
         get-common-topo-info
-          (fn [^String storm-id operation]
-            (let [storm-cluster-state (:storm-cluster-state nimbus)
-                  topology-conf (try-read-storm-conf conf storm-id blob-store)
-                  storm-name (topology-conf TOPOLOGY-NAME)
-                  _ (check-authorization! nimbus
-                                          storm-name
-                                          topology-conf
-                                          operation)
-                  topology (try-read-storm-topology storm-id blob-store)
-                  task->component (storm-task-info topology topology-conf)
-                  base (.storm-base storm-cluster-state storm-id nil)
-                  launch-time-secs (if base (:launch-time-secs base)
-                                     (throw
-                                       (NotAliveException. (str storm-id))))
-                  assignment (.assignment-info storm-cluster-state storm-id nil)
-                  beats (map-val :heartbeat (get @(:heartbeats-cache nimbus)
-                                                 storm-id))
-                  all-components (set (vals task->component))]
-              {:storm-name storm-name
-               :storm-cluster-state storm-cluster-state
-               :all-components all-components
-               :launch-time-secs launch-time-secs
-               :assignment assignment
-               :beats beats
-               :topology topology
-               :task->component task->component
-               :base base}))
+        (fn [^String storm-id operation]
```

Comment:
it will be great if we can preserve previous spaces looks like this diff showing up unnecessarily